### PR TITLE
fix(s3): handle NoSuchBucket error

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Handle NoSuchBucket errors in the following functions:

- GetBucketOwnershipControls
- GetBucketTagging
- GetBucketEncryption
- GetPublicAccessBlock

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/s3/s3_service.py [hunk 1]
```diff
@@ -285,9 +312,14 @@ def __get_bucket_tagging__(self, bucket):
         except ClientError as error:
             bucket.tags = []
             if error.response["Error"]["Code"] != "NoSuchTagSet":
-                logger.error(
-                    f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
-                )
+                if error.response["Error"]["Code"] == "NoSuchBucket":
```
Comment (jfagoagas) [inline prowler/providers/aws/services/s3/s3_service.py:315]: should it be at the same level?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/services/s3/s3_service.py (hunk 1 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/s3/s3_service.py:315]: Nope, we don't want to log the exception when the error is `NoSuchTagSet`

Comment (jfagoagas) [reply prowler/providers/aws/services/s3/s3_service.py:315]: Good one, thanks for the explanation!

**Review by jfagoagas — APPROVED** [review]
🚀

**Pull request merged**